enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -0,0 +1,599 @@
+
+See [Open Questions](#open-questions) for some related discussion.
+
+### Implementation Details/Notes/Constraints
+
+We have several constraints related  design and release-timing constraints to consider.
+
+#### Ownership of the ServiceMeshControlPlane
+
+OpenShift Service Mesh is configured through the ServiceMeshControlPlane CR,
+both for ingress use-cases and for mesh use-cases.  [The existing installation
+procedure for Service
+Mesh](https://docs.openshift.com/container-platform/4.11/service_mesh/v2x/ossm-create-smcp.html#ossm-control-plane-deploy-cli_ossm-create-smcp)
+instructs the reader to create a ServiceMeshControlPlane named "basic" in the
+"istio-system" namespace.  To ensure backwards compatibility, the Ingress
+Operator must do one of the following:
+
+* Use the same name and namespace: "istio-system/basic".
+* Migrate an existing SMCP from "istio-system" to "openshift-ingress".
+* Provide an API to tell the Ingress Operator what SMCP to use.
+
+Placement, possible migration, and ownership of the ServiceMeshControlPlane need
+to be clarified in this enhancement.  This is an open design question.
+
+#### Managing the ServiceMeshMemberRoll
+
+OSSM 2.3 only watches resources in namespaces specified in an
+ServiceMeshMemberRoll CR associated with the ServiceMeshControlPlane; see
+[OSSM-924](https://issues.redhat.com/browse/OSSM-924).  We hope that this
+limitation will be removed from OSSM in time for this enhancement; this is an
+open release-planning question.  Otherwise, the Ingress Operator or cluster
+admin must add each namespace with HTTPRoute objects that Istiod should
+reconcile to the ServiceMeshMemberRoll, and the approach we take and ownership
+of ServiceMeshMemberRoll need to be clarified in this enhancement; this is an
+open design question.
+
+#### Gateway API v1alpha1 CRDs
+
+Istiod fails to become ready if the expected CRDs do not all exist.  OSSM 2.3 is
+based on a version of Istio that expects the Gateway API v1alpha1 CRDs to exist.
+In particular, Gateway API v1alpha1 includes the following CRDs that are not
+defined in Gateway API v1beta1: tcproutes, tlsroutes, and udproutes.  For the
+time being, this means that it is necessary to install the v1alpha1 CRDs in
+order for OSSM 2.3 to function. We hope that OSSM will have a new release based
+on a newer version of Istio in time for this enhancement. This is an open
+release-planning question.
+
+#### Security Policy
+
+TBD.
+
+### Risks and Mitigations
+
+#### Automatic deployments
+
+OpenShift Service Mesh and Istio have a feature called [automated
+deployment](https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#automated-deployment)
+that creates an Envoy deployment and service for each Gateway.  Enabling
+automated deployments and also allowing arbitrary users to create Gateways would
+create a new attack surface for untrusted cluster users.  For this reason, the
+default policy allows only cluster admins to create Gateways.
+
```

Comment:
Another risk is maintaining Release Alignment. We would have to tie features and fixes to specific versions of OSSM if we go to GA without release schedule alignment to OpenShift releases.  E.g. in 4.13 we release with OSSM 2.4, then in OSSM 2.5 a fix is issued that maps to 4.13.5, users will need to upgrade both OpenShift and OSSM.  Backports and downgrades will get complicated as well without release alignment.

Reply:
That's more or less the issue we're struggling with now (we want to base the enhanced dev preview on OSSM 2.4, but doing so requires OSSM 2.4's release to come before OCP 4.13's feature complete date).  I've added some text to describe the issue.